File: Лабораторная работа 1/№2-метод итераций.py
def g(x):
    return math.exp(-x)

def fixed_point_iteration(x0, eps):
    x1 = g(x0)
    while abs(x1 - x0) > eps:
        x0 = x1
        x1 = g(x0)
    return x1

x0 = 1.5
eps = 1e-6

root = fixed_point_iteration(x0, eps)
print(f"Root found at x = {root}")

File: Лабораторная работа 1/№2-метод итераций.cpp
#include <iostream>
#include <cmath>

double g(double x) {
    return 1 / exp(x) + 1;
}

double simpleIteration(double x0, double eps) {
    double x = x0;
    double next_x;
    int iter = 0;

    do {
        next_x = g(x);
        iter++;
        x = next_x;
    } while (std::abs(next_x - x) >= eps);

    return x;
}

int main() {
    double x0 = 2.0;
    double eps = 1e-6;

    double root = simpleIteration(x0, eps);

    std::cout << "Root: " << root << std::endl;

    return 0;
}

